src/peakrdl_sv/regmodel.py

```diff
@@ -22,8 +52,357 @@ def _parse_rdl(self, rdlfile: str) -> AddressMap:
+    ) -> Generator[tuple[Any, int], Any, None]:
+        """Split a value over the fields of a register with correct masking depending
+        on the field location
+
+        :param target: The register to target
+        :type target: Register
+        :param value: The value to write to the registers
+        :type value: int
+        :yield: tuples of the field address and the value to write
+        :rtype: Generator[tuple[Any, int], Any, None]
+        """
+
+        # format the write value an `regwidth`-bit binary string
+        bits = f"{value:0{target.regwidth}b}"
+
+        # Extract the bits we are going to write and conver them to int
+        for field in target:
+            # get the slice indices of the field
+            lower = target.regwidth - field.msb - 1
+            upper = target.regwidth - field.lsb
+
+            # perform the slice (python slicing non inclusive)
+            yield (field.absolute_address, int(bits[lower:upper], 2))  # noqa: E203
+
+    # --------------------------------------------------------------------------------
+    # Generic read and write wrappers
+    # --------------------------------------------------------------------------------
+
+    async def write(
+        self,
+        reg_name: str,
+        field_name: str | None = None,
+        data: int | None = None,
+        **kwargs,
+    ) -> None:
+        """Writes the value of the DUT register. Two options for calling:
+
+        (1) register name + data -> write to named register a particular value
+        (2) register name -> write desired value to a named register
+
+        :param reg_name: Register name to write to
+        :type reg_name: str | int
+        :param data: Data to write to register, defaults to None
+        :type data: int | None, optional
+        """
+
+        if isinstance(field_name, int):
+            self._log.warning(
+                "You passed an int to the `field_name` argument of write(), this is"
+                " probably a mistake, try calling with data=`value`",
+            )
+
+        async def write_desired_to_named_reg(reg_name: str):
+            """Writes a desired register value to the DUT"""
+
+            target = self.get_register_by_name(reg_name)
+
+            # write the register value as multiple `accesswidth` chunks
+            if target.regwidth > target.accesswidth:
+                for field in target:
+                    await self._callbacks.async_write_callback(
+                        field.absolute_address,
+                        self.get_field(reg_name, field.inst_name),
+                        **kwargs,
+                    )
+
+            # one singular write
+            else:
+                await self._callbacks.async_write_callback(
+                    target.absolute_address,
+                    self.get(reg_name),
+                    **kwargs,
+                )
+
+        async def write_desired_to_named_field(reg_name: str, field_name: str):
```

Comment:
What is the intended use case here?  I don't think we should allow writes to a single field as you're going to overwrite all other fields within the access width.  We might want to allow the user to set the desired value on individual fields, but the actual write should be issued from the parent register I think.  Perhaps we need a white board session.

Reply:
> I don't think we should allow writes to a single field as you're going to overwrite all other fields within the access width

Good point, I didn't consider that. I think your suggestion makes sense here but it is an unfortunate limitation that we cant write directly to fields. For registers which are made up of multiple `accesswidth` fields this is nice to have.

In any case, i'll remove this feature ... which also means we can remove the `TypeError` above as no field argument is required.